### PR TITLE
PWGHF: Lc tree creator process switch for data and adjust range histos Lc task

### PR DESCRIPTION
**Pull request by AnnalenaKa**
This modification adds a process switch in the Lc tree creator task. By default it is set to true to process MC. When the switch is set to false the tables are filled for data processing. 
In the Lc task the histogram ranges were adjusted to match the distributions.

### CODE REVIEW

**Comment by AnnalenaKa** [comment]
Hi @strogolo,
here are the proposed changes for the Lc tree creator. I added a process switch that allows to select whether data or MC is processed. Like this it is possible fill the tables with the candidate properties and write them to a tree. If more or other variables should be included in the tree we can still add them.
In the Lc task I adjusted the histogram ranges to match the distributions a bit better.

File: PWGHF/TableProducer/HFTreeCreatorLcToPKPi.cxx [hunk 1]
```diff
@@ -307,6 +307,116 @@ struct CandidateTreeWriter {
...
+            candidate.impactParameter2(),
+            candidate.errorImpactParameter0(),
+            candidate.errorImpactParameter1(),
+            candidate.errorImpactParameter2(),
+            candidate.index0_as<aod::BigTracksPID>().tpcNSigmaPi(),
+            candidate.index0_as<aod::BigTracksPID>().tpcNSigmaKa(),
+            candidate.index0_as<aod::BigTracksPID>().tpcNSigmaPr(),
+            candidate.index0_as<aod::BigTracksPID>().tofNSigmaPi(),
+            candidate.index0_as<aod::BigTracksPID>().tofNSigmaKa(),
+            candidate.index0_as<aod::BigTracksPID>().tofNSigmaPr(),
```
Comment (vkucera) [inline PWGHF/TableProducer/HFTreeCreatorLcToPKPi.cxx:386]: Avoid multiple calls of `index0_as`. It is expensive.

**Review by vkucera — CHANGES_REQUESTED** [review]

File: PWGHF/TableProducer/HFTreeCreatorLcToPKPi.cxx (hunk 1 above)
Comment (AnnalenaKa) [reply PWGHF/TableProducer/HFTreeCreatorLcToPKPi.cxx]: Dear Vit,
thanks for pointing this out. I replaced this by calling it once before the table is filled with the properties.

File: PWGHF/TableProducer/HFTreeCreatorLcToPKPi.cxx [hunk 2]
```diff
@@ -307,6 +310,119 @@ struct CandidateTreeWriter {
...
+      auto fillTable = [&](int CandFlag,
+                           int FunctionSelection,
+                           float FunctionInvMass,
+                           float FunctionCt,
+                           float FunctionY,
+                           float FunctionE) {
+        if (FunctionSelection >= 1) {
+          rowCandidateFull(
+            candidate.index0_as<aod::BigTracksPID>().collision().bcId(),
+            candidate.index0_as<aod::BigTracksPID>().collision().numContrib(),
```
Comment (vkucera) [inline PWGHF/TableProducer/HFTreeCreatorLcToPKPi.cxx:348]: Same here

**Review by vkucera — CHANGES_REQUESTED** [review]

**Review by vkucera — APPROVED** [review]

**Pull request merged**